depscan/lib/utils.py

```diff
@@ -257,9 +257,21 @@ def search_pkgs(project_type: str | None, pkg_list: List[Dict[str, Any]]):
                     purl_aliases[f"{vari_full_pkg.lower()}:{version}"] = pkg.get("purl")
     raw_results = []
     for i in expanded_list:
-        search_term = i.get("purl") or i.get("name")
-        if res := search_by_purl_like(search_term, with_data=True):
-            raw_results.extend(res)
+        if i.get("purl"):
```

Comment:
@prabhu Can we just modify line 260 to be `i.get("purl") or i.get("cpe") or i.get("url")` then just do `if search_term` and use `search_by_any` instead?

Reply:
Sounds like a good idea. Could you kindly update this branch with your refactoring?